publication-file/src/main/java/no/unit/nva/publication/file/upload/FileService.java

```diff
@@ -56,7 +80,37 @@ public InitiateMultipartUploadResult initiateMultipartUpload(SortableIdentifier
         return amazonS3.initiateMultipartUpload(request);
     }
 
-    public boolean customerDoesNotAllowUploadingFile(Customer customer, Resource resource) {
+    public UploadedFile completeMultipartUpload(SortableIdentifier resourceIdentifier,
+                                                CompleteUploadRequestBody completeUploadRequestBody,
+                                                UserInstance userInstance) throws NotFoundException {
+
+        var resource = Resource.resourceQueryObject(resourceIdentifier).fetch(resourceService)
+            .orElseThrow(() -> new NotFoundException(RESOURCE_NOT_FOUND_MESSAGE));
+
+        var completeMultipartUploadRequest = completeUploadRequestBody.toCompleteMultipartUploadRequest(BUCKET_NAME);
+        var completeMultipartUploadResult = amazonS3.completeMultipartUpload(completeMultipartUploadRequest);
+        var s3ObjectKey = completeMultipartUploadResult.getKey();
+        var objectMetadata = getObjectMetadata(s3ObjectKey);
+
+        var file = constructUploadedFile(s3ObjectKey, objectMetadata, userInstance);
+
+        FileEntry.create(file, resource.getIdentifier(), userInstance).persist(resourceService);
+
+        return file;
+    }
+
+    private ObjectMetadata getObjectMetadata(String key) {
+        return amazonS3.getObjectMetadata(new GetObjectMetadataRequest(BUCKET_NAME, key));
+    }
+
+    private UploadedFile constructUploadedFile(String identifier, ObjectMetadata objectMetadata,
+                                               UserInstance userInstance) {
+        return new UploadedFile(UUID.fromString(identifier), toFileName(objectMetadata.getContentDisposition()),
```

Comment:
maybe move the fromString one level out and accept UUID instead to keep stuff a bit more reusable